### PR TITLE
Segmentmanager fill with paintable features

### PR DESCRIPTION
**Pull request by LorenzLamm**
Adjusted the _fill_ functionality analogously to the _paint_ function to not fill labels that are not paintable.

### CODE REVIEW

File: src/cryocanvas/fill.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        # If requested new label doesn't change old label then return
+        # If preserve_labels is True, then only paint over paintable labels
+        old_label = np.asarray(self.data[int_coord]).item()
+        # get the paintable label values
+        features_table = self.features
+        paintable_values = features_table.loc[features_table[PAINTABLE_KEY]]["index"].values
+        if old_label == new_label or (
+            self.preserve_labels
+            and old_label not in paintable_values
```
Comment (LorenzLamm) [inline src/cryocanvas/fill.py:41]: As suggested by @kevinyamauchi , I only adjusted this line to check whether the selected instance is  paintable. If it is not, nothing will be filled.

**Comment by LorenzLamm** [comment]
In the following example, the brown class is locked, and the purple one _paintable_.
When clicking at the background within the circled area, filling will do this:
fill_snap1fill_snap2

Side note:
Would it make sense to also have functionality that overwrites the _paintable_ features here?
fill_snap3
Maybe something like this could be useful when a user would like to filter out a lot of small connected components in an area. But probably there are better ways of getting there...

File: src/cryocanvas/_tests/test_segment_manager.py [hunk 2]
```diff
@@ -80,3 +80,66 @@ def test_feature_validation_valid_features_table():
         class_values,
         features_table[CLASS_KEY]
     )
+
+
+def test_paint_with_locked_features():
```
Comment (LorenzLamm) [inline src/cryocanvas/_tests/test_segment_manager.py:85]: Also added some tests here for the painting / filling functions.
Any feedback welcome :)

Comment (LorenzLamm) [reply src/cryocanvas/_tests/test_segment_manager.py:85]: Many thanks for the feedback @kevinyamauchi :)
Not sure I understand exactly, though. You mean to not only test whether locked features are locked, but also paintable features are painted?

In the most recent commit, I added checking for concrete values (see below). Is that what you mean?

File: src/cryocanvas/_tests/test_segment_manager.py [hunk 3]
```diff
@@ -80,3 +80,70 @@ def test_feature_validation_valid_features_table():
...
+
+    # paint over a locked feature
+    labels_layer.paint((1, 1, 1), 2)
+
+    # check that the locked feature was not painted over
+    np.testing.assert_array_equal(
+        labels_layer.data, _make_labels_layer().data
+    )
+
+    # paint over a paintable feature
```
Comment (LorenzLamm) [inline src/cryocanvas/_tests/test_segment_manager.py:113]: Here is the check for paintable features

**Review by kevinyamauchi — APPROVED** [review]
Thanks for the quick update, @LorenzLamm ! You understood correctly. I wanted the test to check the values to make sure they were updated to the correct value.

I think this looks good. I will merge now. Thanks!

File: src/cryocanvas/_tests/test_segment_manager.py (hunk 3 above)
Comment (kevinyamauchi) [reply src/cryocanvas/_tests/test_segment_manager.py:113]: Perfect, thanks! This is totally fine. Just for your future reference, there are useful array testing functions from numpy that may be useful: https://numpy.org/doc/stable/reference/routines.testing.html

**Pull request merged**